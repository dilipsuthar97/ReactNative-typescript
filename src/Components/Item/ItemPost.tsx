import * as React from 'react';
import {
	View,
	Text,
	StyleSheet,
	TouchableOpacity,
	StyleProp,
	ViewStyle,
} from 'react-native';

import { Scale, Colors, Matrics } from '../../CommonConfig';
import { Post } from '../../Types/Post.interface';

interface Props {
	item: Post;
	style?: StyleProp<ViewStyle>;
	onPress?: () => void;
}

const ItemPost: React.FC<Props> = ({ item, style, onPress }) => {
	return (
		<TouchableOpacity
			style={[styles.container, style]}
			activeOpacity={0.5}
			onPress={onPress}
		>
			<Text style={styles.title} numberOfLines={2}>
				{item ? item.title : undefined}
			</Text>
			<Text style={styles.body} numberOfLines={3}>
				{item ? item.body : undefined}
			</Text>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	container: {
		padding: Scale(8),
		borderRadius: Scale(8),
		shadowColor: Colors.BLACK,
		shadowRadius: Scale(8),
		shadowOffset: { width: 0.2, height: 0.2 },
		shadowOpacity: 0.2,
		backgroundColor: Colors.WHITE,
		marginHorizontal: Scale(10),
	},
	title: {
		fontSize: Scale(14),
		fontWeight: 'bold',
		color: Colors.BLACK,
	},
	body: {
		fontSize: Scale(14),
		color: Colors.MATEBLACK,
	},
});

export default ItemPost;
