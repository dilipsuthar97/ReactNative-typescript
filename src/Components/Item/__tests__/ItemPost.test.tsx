import React from 'react';
import renderer from 'react-test-renderer'

import ItemPost from '../ItemPost';

it('render correctly with defaults', () => {
    const tree = renderer.create(<ItemPost item={{id: 100, title: 'title'}}/>).toJSON();
    expect(tree).toMatchSnapshot();
})