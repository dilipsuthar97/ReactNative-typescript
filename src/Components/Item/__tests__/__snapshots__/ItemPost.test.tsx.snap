// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly with defaults 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 16,
      "marginHorizontal": 20,
      "opacity": 1,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0.2,
        "width": 0.2,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 16,
    }
  }
>
  <Text
    numberOfLines={2}
    style={
      Object {
        "color": "#000",
        "fontSize": 27,
        "fontWeight": "bold",
      }
    }
  >
    title
  </Text>
  <Text
    numberOfLines={3}
    style={
      Object {
        "color": "#343434",
        "fontSize": 27,
      }
    }
  />
</View>
`;
